lib/connection/pool.js

```diff
@@ -783,10 +790,9 @@ Pool.prototype.connect = function() {
  * @param {...object} param Parameters for the specific mechanism
  * @param {authResultCallback} callback A callback function
  */
-Pool.prototype.auth = function(mechanism) {
+Pool.prototype.auth = function(credentials, callback) {
   var self = this;
-  var args = Array.prototype.slice.call(arguments, 0);
-  var callback = args.pop();
+  const mechanism = credentials ? credentials.mechanism : '';
```

Comment:
seems to me we should rather bail if no `credentials` were provided? Isn't there always a mechanism, i.e. `default` if none was provided

Reply:
We do not validate this far down. TBH anything that isn't a valid authprovider should have errored either in native or in the server by the time it gets here.